mne_sandbox/connectivity/cfc.py

```diff
@@ -121,17 +122,136 @@ def phase_amplitude_coupling(inst, f_phase, f_amp, ixs, pac_func='ozkurt',
+        the values. Function must accept an N-D input and will operate on the
+        last dimension. E.g., `sklearn.preprocessing.scale`.
+        Defaults to no scaling.
+    n_jobs : int
+        Number of jobs to run in parallel. Defaults to 1.
+    """
+    def __init__(self, f_phase, f_amp, ixs, sfreq, times,
+                 pac_func='ozkurt', tmin=0., tmax=1., n_cycles_ph=3,
+                 n_cycles_am=3, scale_amp_func=None, n_jobs=1):
+        events, times, tmin, tmax = _check_pac_times(None, times, tmin, tmax)
+        (ixs, f_phase, f_amp, n_cycles_ph,
+         n_cycles_am, pac_func) = _check_pac_data(
+            ixs, f_phase, f_amp, n_cycles_ph, n_cycles_am, pac_func)
+
+        self.f_phase = f_phase
+        self.f_amp = f_amp
+        self.tmin = tmin
+        self.tmax = tmax
+        self.times = times
+        self.ixs = ixs
+        self.sfreq = sfreq
+        self.pac_func = pac_func
+        self.n_cycles_ph = n_cycles_ph
+        self.n_cycles_am = n_cycles_am
+        self.scale_amp_func = scale_amp_func
+        self.n_jobs = n_jobs
+
+    def fit(self, X=None, y=None):
+        """
+        Parameters
+        ----------
+        X : None | array
+        y : None | array
+        """
+        return self
+
+    def transform(self, X, y=None):
+        """Estimate PAC values.
+
+        Parameters
+        ----------
+        X : array, shape (n_channels, n_times) |
+                (n_epochs, n_channels, n_times)
+            The input data.
+        y : None | array
+        """
+        n_ep = X.shape[0]
+        n_f_pairs = len(self.f_amp) * len(self.f_phase)
+        n_t_wins = len(self.tmin)
+        n_ch_pairs = len(self.ixs)
+
+        pac = np.zeros([X.shape[0], n_f_pairs * n_t_wins * n_ch_pairs])
+        for ii, ep_X in enumerate(X):
+            # Output will be shape (n_epochs, n_ch_pair, n_fr_pair, n_ti_win)
+            i_pac = _phase_amplitude_coupling(
+                ep_X, self.sfreq, self.f_phase, self.f_amp, self.ixs,
+                self.pac_func, None, self.tmin, self.tmax, self.times,
+                self.n_cycles_ph, self.n_cycles_am, self.scale_amp_func)[0]
+            # Reshape PAC so it's 2D and first dimension is Epochs
+            pac[ii, :] = i_pac.reshape(1, n_f_pairs * n_t_wins * n_ch_pairs)
+        return pac
+
+    def fit_transform(self, X, y=None, times=None):
+        """Estimate PAC values.
+
+        Parameters
+        ----------
+        X : array, shape (n_channels, n_times) |
+                (n_epochs, n_channels, n_times)
+            The input data.
+        y : None | array
+        """
+        return self.transform(X)
+
+
 def _phase_amplitude_coupling(data, sfreq, f_phase, f_amp, ixs,
                               pac_func='ozkurt', events=None,
-                              tmin=None, tmax=None, n_cycles_ph=3,
+                              tmin=None, tmax=None, times=None, n_cycles_ph=3,
```

Comment:
are tmin and tmax necessary for epochs?

I think it's easier just to add a step in the pipeline e.g. `make_pipeline(TimeSelector(), PAC())` to reduce number of params